back-end/service.py

```diff
@@ -131,8 +133,8 @@ def updateSecret(secret, data, user, repo):
         log('debug', 'Secret updated in `{}`'.format(repo))
         return True
     else:
-        raise ServerError('Secret could not be updated in `{}`'.format(repo), 412)
-        log('critical', 'Execution may not work properly from here.')
+        raise ServerError('Secret could not be updated in `{}`'.format(repo),
+            HTTPStatus(417))
```

Comment:
Same as the one above, just return `500`.

Reply:
Changed and pushed.